From 50a3a22319f66b334170dd183d276843407bf2ab Mon Sep 17 00:00:00 2001
From: Vivek Kasireddy <vivek.kasireddy@intel.com>
Date: Fri, 27 Apr 2018 10:48:33 -0700
Subject: [PATCH 0915/1240] INTERNAL [IOTG] drm/i915: Check initial modeset in
 max_available_pipes

During testing of the planes restriction feature, it was discovered that
the series would fail to boot under LaaG and AaaG. Vivek offered this
commit as a fix. With this commit, AaaG will boot to GFX without
initial modeset being passed to the kernel.


This was also tested to work on AaaG and LaaG with these changs to
the command line paramaters:
i915.enable_initial_modeset=1 and i915.avail_planes_per_pipe=0x000C00

Fixes: faa158d ("INTERNAL [IOTG] drm/i915: Introduce the Plane Restriction feature")

Change-Id: I36850b21404767aa1a3bb544bc407b0533347941
Signed-off-by: Brian J Lovin <brian.j.lovin@intel.com>
Signed-off-by: Vivek Kasireddy <vivek.kasireddy@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 46de16a..84ff28a 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1314,7 +1314,8 @@ static inline int get_max_avail_pipes(struct drm_i915_private *dev_priv)
 	int index = 0;
 
 	if (!intel_vgpu_active(dev_priv) ||
-	    !i915_modparams.avail_planes_per_pipe)
+	    !i915_modparams.avail_planes_per_pipe ||
+	    !i915_modparams.enable_initial_modeset)
 		return INTEL_INFO(dev_priv)->num_pipes;
 
 	for_each_pipe(dev_priv, pipe) {
-- 
2.7.4

